src/miniscript/decode.rs

```diff
@@ -34,6 +35,54 @@ fn return_none<T>(_: usize) -> Option<T> {
     None
 }
 
+/// Trait for parsing keys from byte slices
+pub trait ParseableKey: Sized + private::Sealed {
+    /// Parse a key from slice
+    fn from_slice(sl: &[u8]) -> Result<Self, KeyParseError>;
+}
+
+/// Decoding error while parsing keys
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum KeyParseError {
+    /// Bitcoin PublicKey parse error
+    FullKeyParseError(bitcoin::util::key::Error),
+    /// Xonly key parse Error
+    XonlyKeyParseError(bitcoin::secp256k1::Error),
+}
+
+impl ParseableKey for bitcoin::PublicKey {
+    fn from_slice(sl: &[u8]) -> Result<Self, KeyParseError> {
+        bitcoin::PublicKey::from_slice(sl).map_err(KeyParseError::FullKeyParseError)
+    }
+}
+
+impl ParseableKey for bitcoin::schnorr::PublicKey {
+    fn from_slice(sl: &[u8]) -> Result<Self, KeyParseError> {
+        bitcoin::schnorr::PublicKey::from_slice(sl).map_err(KeyParseError::XonlyKeyParseError)
+    }
+}
+
+impl error::Error for KeyParseError {}
```

Comment:
Better to implement `source` methods and return inner error type here

Reply:
source is MSRV 1.30 :( . I will make an issue for better error handing to change everything to source. I can use the now deprecated `cause` for now, will make an issue to change everything to source once we change MSRV

Reply:
Oh yes. Ok, probably it is better not to change than and have an issue for `source` adoption once we will bump MSRV